fresh_setup.sh

```diff
@@ -1,96 +1,110 @@
 #!/bin/bash
 
+# TODO:
+# - look into `apt install unattended-upgrades`
+# - look into activating fail2ban for ssh protection (I doubt we'll need this unless we get too much random traffic)
+
 # this is a script for seting up the website from a fresh install
 
 echo "hi sysadmin!"
 echo "this script will install (almost) everything needed to run the csss website"
-echo "==="
-
-echo "update apt to latest packages..."
-sudo apt update && sudo apt upgrade -y
-
-# TODO: look into `sudo apt install unattended-upgrades`
-# TODO: look into activating fail2ban for ssh protection (I doubt we'll need this unless we get too much random traffic)
-
-echo "installing git..."
-sudo apt install git
-
-echo "creating csss_site user account..."
-sudo useradd csss-site -m # has home
-sudo usermod -L csss-site # cannot login
-
-echo "installing python3.12"
-sudo apt-get install software-properties-common
-sudo apt update
-sudo apt install python3.11 python3.11-venv -y # default on debian 12
-
-echo "installing supervisor & nginx"
-sudo apt install nginx -y
-#chmod g=rwx /home/csss-site/csss-site-backend/src -R
-
-echo "clone csss-site backend"
+echo "(make sure you are running on a Debian 12 Linux machine as the superuser!)"
+
+# ask the user for consent to proceed
+while true; do
+	echo "(P)roceed, (c)ancel?"
+	read choice
+
+	if [ $choice = 'P' ]; then
+		break
+	elif [ $choice = 'c' ]; then
+		exit 0
+	else
+		echo "Not sure what you mean..."
+	fi
+done
+
+echo "----"
+echo "configure apt sources..."
+echo "deb https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+
+echo "----"
+echo "update and upgrade apt..."
+apt update && apt upgrade -y
+
+echo "----"
+echo "install packages..."
+apt install git software-properties-common python3.11 python3.11-venv libaugeas0 nginx postgresql-15 postgresql-contrib -y
+# install certbot
+python3 -m venv /opt/certbot
+/opt/certbot/bin/pip install --upgrade pip
+/opt/certbot/bin/pip install certbot certbot-nginx
+ln -s /opt/certbot/bin/certbot /usr/bin/certbot
+
+echo "----"
+echo "add user csss_site..."
+useradd csss-site -m # -m: has home /home/csss-site
+usermod -L csss-site # -L: cannot login
+chsh -s /usr/bin/bash csss-site # make user csss-site use the bash shell
 cd /home/csss-site
-git clone git@github.com:CSSS/csss-site-backend.git
 
-echo "creating a virtual environment for python"
-python3.11 -m venv .venv
+echo "----"
+echo "clone repository csss-site-backend..."
+sudo -u csss-site git clone https://github.com/CSSS/csss-site-backend.git csss-site-backend
+
+echo "----"
+echo "configure sudo for csss-site..."
+cp csss-site-backend/config/sudoers.conf /etc/sudoers.d/csss-site
+
+echo "----"
+echo "configure nginx..."
+cp csss-site-backend/config/nginx.conf /etc/nginx/sites-available/csss-site-backend
+echo "You'll need to fill out the certbot configuration manually."
```

Comment:
could you link to documents describing how the config should be filled out?

Reply:
100%! Also, I'm working on this part right now, as certbot doesn't configure our config. It configures /etc/nginx/sites-enabled/default. So I'm going to try and play with our config and remove default so it actually pastes information into ours.

Reply:
ah! I see. Do be careful that anything automatic isn't checked into source code. Or if it is, certbot knows not to overwrite it.